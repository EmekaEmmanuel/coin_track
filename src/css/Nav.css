* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.app_container {
  margin: auto;
  background-color: #111315;
}

.home_header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 0;
  background-color: #111315;
  width: 90%;
  margin: auto;
}

.logo_box {
  display: flex;
  align-items: center;
  gap: 5px;
}

.hamburger_container {
  display: none;
}

.logo_img {
  width: 50px;
  object-position: center;
  object-fit: cover;
}

.nav_tag {
  display: flex;
  list-style: none;
  gap: 20px;
}

.nav_font2 {
  color: #efefef;
}

.nav_item {
  padding: 7px 10px;
  border-radius: 10px;
}

.notify_box {
  display: flex;
  gap: 15px;
}

.notify_box img {
  width: 18px;
  height: 18px;
  object-fit: contain;
  object-position: center;
}

.nav_font1 {
  color: #efefef;
}

.active {
  background-color: #373b3e;
}

.td_none {
  text-decoration: none;
}

.vertical_line {
  border-right: 1px solid #e8e8e8;
}

.border_none {
  border: none;
}

.outline_none {
  outline: none;
}

.header_line {
  border-bottom: 0.5px solid #efefef9a;
}

@media only screen and (max-width: 768px) {
  .nav_tag {
    display: flex;
    list-style: none;
    gap: 10px;
  }
}

@media only screen and (max-width: 425px) {
  .home_header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 20px 0;
    background-color: #111315;
    width: 90%;
    margin: auto;
  }

  .logo_box {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    gap: 95px;
  }

  .disp_none {
    display: none;
  }
}

@media only screen and (max-width: 425px) {
  .home_header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 20px 0 10px 0;
    background-color: #111315;
    width: 90%;
    margin: auto;
  }

  .logo_box {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    gap: 80px;
  }
}
