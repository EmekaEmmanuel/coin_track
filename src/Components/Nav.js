import React from 'react';
import '../css/Nav.css';
import { NavLink } from 'react-router-dom';
import closePopUp from '../img/closepopup.png';
import hamburger from '../img/hamburger.png';
import cryptoImage from '../img/crypto_image.png';

function Nav() {
  return (
    <div className="app_container">

      <header className="home_header_box header_line">

        <div className="home_header ">
          <div className="logo_box">
            <figure>
              <img className="logo_img" src={cryptoImage} alt="" />
            </figure>
            <h2 className="nav_font1">monie track</h2>
          </div>

          <nav className="nav_tag">
            <li className="active nav_item"><NavLink to="/" activeClassName="active" className="nav_font2 td_none">Track </NavLink></li>
            <li className="active nav_item"><NavLink to="/" activeClassName="active" className="nav_font2 td_none">Happenings </NavLink></li>
            <li className="active nav_item"><NavLink to="/ChartPage" activeClassName="active" className="nav_font2 td_none">Chart </NavLink></li>
          </nav>

          <div className="notify_box">
            <button type="button">
              {' '}
              <img src={closePopUp} alt="" />
            </button>
            <button type="button">
              {' '}
              <img src={closePopUp} alt="" />
            </button>
          </div>

          <div className="hamburger_container">
            <figure>
              <img src={hamburger} alt="Hamburger_menu" />
            </figure>
          </div>
        </div>

      </header>
    </div>
  );
}

export default Nav;
