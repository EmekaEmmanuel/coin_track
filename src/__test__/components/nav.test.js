import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../../Components/Nav';

it('Navigation component renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
