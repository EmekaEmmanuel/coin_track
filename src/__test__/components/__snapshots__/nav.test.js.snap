// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<div
  className="app_container"
>
  <header
    className="home_header_box header_line"
  >
    <div
      className="home_header "
    >
      <div
        className="logo_box"
      >
        <figure>
          <img
            alt=""
            className="logo_img"
            src="crypto_image.png"
          />
        </figure>
        <h2
          className="nav_font1"
        >
          monie track
        </h2>
      </div>
      <nav
        className="nav_tag"
      >
        <li
          className="active nav_item"
        >
          <a
            activeClassName="active"
            aria-current="page"
            className="nav_font2 td_none active"
            href="/"
            onClick={[Function]}
          >
            Track 
          </a>
        </li>
        <li
          className="active nav_item"
        >
          <a
            activeClassName="active"
            aria-current="page"
            className="nav_font2 td_none active"
            href="/"
            onClick={[Function]}
          >
            Happenings 
          </a>
        </li>
        <li
          className="active nav_item"
        >
          <a
            activeClassName="active"
            className="nav_font2 td_none"
            href="/ChartPage"
            onClick={[Function]}
          >
            Chart 
          </a>
        </li>
      </nav>
      <div
        className="notify_box"
      >
        <button
          type="button"
        >
           
          <img
            alt=""
            src="closepopup.png"
          />
        </button>
        <button
          type="button"
        >
           
          <img
            alt=""
            src="closepopup.png"
          />
        </button>
      </div>
      <div
        className="hamburger_container"
      >
        <figure>
          <img
            alt="Hamburger_menu"
            src="hamburger.png"
          />
        </figure>
      </div>
    </div>
  </header>
</div>
`;
